src/components/TimelineComponent.tsx

```diff
@@ -475,7 +507,51 @@ export default class TimelineComponent
         .attr('class', 'plot')
         .attr('id', 'bars');
 
+    // Column labels
+    this.svg.append('text')
+        .attr('transform',
+            'translate(' + (width/2) + ' ,' +
+                      (height + marginTop + 20) + ')')
+        .style('text-anchor', 'middle')
+        .text(this.state.xColumn);
+
+    this.svg.append('text')
+        .attr('transform', 'rotate(-90)')
+        .attr('y', 0 - marginLeft)
+        .attr('x', 0 - (height / 2))
+        .attr('dy', '1em')
+        .style('text-anchor', 'middle')
+        .text(this.state.yColumn);
+
     this.updateBars();
+
+    // Handle keypresses
+    d3.select('body')
+        .on('keypress', () => {
+          if (d3.event.key === '-' || d3.event.key === 's') {
+            const identity = d3.zoomIdentity
+                .scale(Math.max(scaleMin, this.scale * scaleZoomOut))
+                .translate(1, 1);
+
```

Comment:
Some in-line commenting would be appreciated. For example defining what an 'identity' is tracking.